packages/client/src/interceptors.ts

```diff
@@ -81,6 +98,13 @@ export interface WorkflowClientInterceptor {
    * {@link signalWithStart} most likely needs to be implemented too
    */
   start?: (input: WorkflowStartInput, next: Next<this, 'start'>) => Promise<string /* runId */>;
+  /**
+   * Intercept a service call to updateWorkflowExecution
+   */
+  startUpdate?: (
+    input: WorkflowStartUpdateInput,
+    next: Next<this, 'startUpdate'>
+  ) => Promise<WorkflowStartUpdateOutput>;
```

Comment:
This is the client interceptor we were discussing. I've given the interceptor and the custom object returned names that remind the user that it is not the update in general but specifically `startUpdate` that they are intercepting. This helps a bit with the potential mistake a user might make wherein they take an action on results coming back by this code path but forget about results obtained via polling.